interfaces/sourcegen/run.py

```diff
@@ -0,0 +1,7 @@
+# This file is part of Cantera. See License.txt in the top-level directory or
+# at https://cantera.org/license.txt for license and copyright information.
```

Comment:
For the C# stuff at least, I want the language-specific builder (in this case MSBuild d/b/a dotnet CLI), to be able to shell out to the script. This is because the source files are generated in a temporary build folder called `./obj` which is again deeply-tied to the .NET build process.

You can see the magic in Cantera.csproj line 44 through 56. Line 50 shells out to script if it notices the generated source directory is missing or we force it by specifying an command line option. Line 52 and 53 tell the system to include the generated files in the compilation process.

There is yet more magic! The [OmniSharp](http://www.omnisharp.net/) tooling, which underpins the C# plugins in VSCode, VIM, Sublime, etc., knows how to process the csproj file, and by including the source generation there, all the generated symbols are available for code completion and other static analysis.

So... I don’t have a huge preference where the sourcegen stuff lives, but I do want to make sure we can call it outside SCons. That said, I do see intergrating the .NET package build with Scons using a Command or custom builder to orchestrate the [.NET CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/)

Reply:
> So... I don’t have a huge preference where the sourcegen stuff lives, but I do want to make sure we can call it outside SCons. That said, I do see intergrating the .NET package build with Scons using a Command or custom builder to orchestrate the [.NET CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/)

I don’t have much to add other than that the final build process should go through Scons, which is what any other compilations from source are currently using. Ideally, everything beyond SCons should be based on auto-detection of tools within an existing build environment.

Reply:
I think there's some simplicity to be had in keeping `sourcegen` independent of SCons, and would rather avoid mingling it with the stuff in `site_scons`, at least for the time being.

While for other generation tasks, it seems likely that SCons would call `sourcegen` directly, I think @burkenyo's point here is that for C#, we should expect that ultimately, SCons will call `msbuild` and `msbuild` should call `sourcegen`, though I don't think the first part of that integration needs to be resolved as part of this PR.